tests/nested/manual/core20-set-efi-boot-vars/task.yaml

```diff
@@ -0,0 +1,220 @@
+
+details: >
+    This test checks that EFI boot variables are correctly set during
+    installation.  In particular, there should be a Boot#### variable with
+    data pointing to either \EFI\ubuntu\shimx64.efi or \EFI\boot\bootx64.efi,
+    depending on the version of the pc-gadget snap, and the first number in the
+    BootOrder variable should be the corresponding number.
+
+systems:
+  - -ubuntu-1*
+
+environment:
+    DESCRIPTION/BOOTDIR: "Install with assets in EFI/boot/, then update to gadget with assets in EFI/ubuntu/"
+    DESCRIPTION/UBUNTUDIR: "Install with assets in EFI/ubuntu/, then update to gadget with assets in EFI/boot/"
+    DESCRIPTION/BOOTDIR_NOSEC: "Install with assets in EFI/boot/, then update to gadget with assets in EFI/ubuntu/ (no secure boot)"
+    DESCRIPTION/UBUNTUDIR_NOSEC: "Install with assets in EFI/ubuntu/, then update to gadget with assets in EFI/boot/ (no secure boot)"
+
+    INITIAL_ASSET_DIR/BOOTDIR: "boot"
+    INITIAL_ASSET_DIR/UBUNTUDIR: "ubuntu"
+    INITIAL_ASSET_DIR/BOOTDIR_NOSEC: "boot"
+    INITIAL_ASSET_DIR/UBUNTUDIR_NOSEC: "ubuntu"
+
+    FINAL_ASSET_DIR/BOOTDIR: "ubuntu"
+    FINAL_ASSET_DIR/UBUNTUDIR: "boot"
+    FINAL_ASSET_DIR/BOOTDIR_NOSEC: "ubuntu"
+    FINAL_ASSET_DIR/UBUNTUDIR_NOSEC: "boot"
+
+    NESTED_ENABLE_TPM/BOOTDIR: "true"
+    NESTED_ENABLE_TPM/UBUNTUDIR: "true"
+    NESTED_ENABLE_TPM/BOOTDIR_NOSEC: "false"
+    NESTED_ENABLE_TPM/UBUNTUDIR_NOSEC: "false"
+
+    NESTED_ENABLE_SECURE_BOOT/BOOTDIR: "true"
+    NESTED_ENABLE_SECURE_BOOT/UBUNTUDIR: "true"
+    NESTED_ENABLE_SECURE_BOOT/BOOTDIR_NOSEC: "false"
+    NESTED_ENABLE_SECURE_BOOT/UBUNTUDIR_NOSEC: "false"
+
+prepare: |
+    snap install yq
+
+    VERSION=$(tests.nested show version)
+    echo "Download pc-gadget to use in initial image"
+    snap download --basename=pc --channel="${VERSION}/${GADGET_CHANNEL}" pc
+    unsquashfs -d pc-gadget pc.snap
+
+    ARCH=$(find pc-gadget -name 'grub*.efi' -printf '%f\n' | sed 's/grub//;s/\.efi//')
+
+    case "${ARCH}" in
+        x64  ) ;;
+        aa64 ) ;;
+        *    ) ERROR "Invalid architecture '${ARCH}': must be 'x64' or 'aa64'" ;;
+    esac
+
+    echo "Set up gadget with EFI assets in EFI/${INITIAL_ASSET_DIR}"
+    case "${INITIAL_ASSET_DIR}" in
+        boot )
+            sh modify-gadget.sh pc-gadget "$ARCH" no-fallback
+            ;;
+        ubuntu )
+            sh modify-gadget.sh pc-gadget "$ARCH" fallback
+            ;;
+        * )
+            ERROR "Invalid initial asset dir: ${INITIAL_ASSET_DIR}"
+            ;;
+    esac
+
+    if [ "${INITIAL_ASSET_DIR}" = "ubuntu" ]; then
+      # make the entry different for fallback
+      sed -i 's/,ubuntu,/,fallback-created,/g' "pc-gadget/BOOT${ARCH^^}.CSV"
+    fi
+
+    echo "Get snakeoil key"
+    KEY_NAME=$(tests.nested download snakeoil-key)
+    SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+    SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+    echo "Sign the modified pc-gadget"
```

Comment:
nitpick^3: strictly speaking we are signing shim in the gadget or maybe other assets